OpenRA.Game/GameRules/WeaponInfo.cs

```diff
@@ -33,6 +33,27 @@ public class ProjectileArgs
 		public Target GuidedTarget;
 	}
 
+	public class WarheadArgs
+	{
+		public WeaponInfo Weapon;
+		public int[] DamageModifiers;
```

Comment:
This needs to be `public int[] DamageModifiers = { };` to avoid a crash when firing the ion cannon (and probably other things - the ion cannon was just the first thing I tested).